internal/app/synchronizer/evaluator_client.go

```diff
@@ -145,6 +147,10 @@ func (ec *grcpEvaluatorClient) evaluate(ctx context.Context, pc <-chan []*pb.Mat
 			if err != nil {
 				return fmt.Errorf("failed to get response from evaluator client, desc: %w", err)
 			}
+			if !matchIDs[resp.GetMatchId()] {
+				return fmt.Errorf("evaluator returned unmatched/duplicated matchID %s which does not correspond to its input", resp.GetMatchId())
+			}
+			matchIDs[resp.GetMatchId()] = false
```

Comment:
We should have e2e tests for this.

Reply:
Though we don't have a great way to set up a custom evaluator, hrm.

Reply:
I'll mark this as a TODO.

Reply:
Use issues, not code todos.